src/lib/schedule.c

```diff
@@ -187,14 +187,14 @@ static struct task *schedule_edf(void)
 
 	tracev_pipe("edf");
 
+	interrupt_clear(PLATFORM_SCHEDULE_IRQ);
```

Comment:
Can you check master for a similar patch by @tlauda and backport.

Reply:
I did not see any similar patch.
@tlauda 
Do you make some change like this in master?

Reply:
Can you change the commit message, the real fix here is to move irq_clear before edf_get_next as there is a race when adding a new task.